ansible/roles/deploy-baremetal/tasks/prepare.yaml

```diff
@@ -0,0 +1,65 @@
+- name: Allow DHCP and HTTP services in the firewall
+  firewalld:
+    service={{item}} state=enabled
+    permanent=False immediate=True
+  with_items:
+    - http
+    - dhcp
+
+- name: Define DHCP parameters for the target node
+  template:
+    src: dhcp-host
+    dest: /etc/dnsmasq.d/{{ baremetal.ip_address }}
+    owner: root
+    group: root
+    mode: 0644
+
+# machines booting via PXE expect configuration files in some standard
+# places like:
+# pxelinux.cfg/01-ma-ca-dd-re-ss
+# pxelinux.cfg/default
+# more info: http://jk.ozlabs.org/blog/post/158/netbooting-petitboot/
+- name: Create PXE configuration file (network boot) for the target node
+  template:
+    src: pxeconfig-network
+    dest: "{{ http_server.root }}/pxelinux.cfg/01-{{ baremetal.mac_address | regex_replace(':', '-') }}"
+    owner: root
+    group: root
+    mode: 0644
+
+- name: Set machine_dir variable
+  set_fact:
+    machine_dir: "{{ http_server.root }}/{{ baremetal.mac_address }}"
```

Comment:
Can't this variable be set in `deplay-baremetal/defaults/main.yaml`?
Also, you're using the `http_server` variable from `baremetal-ctrl/defaults/main.yaml`, shouldn't it be in a "global" vars file, or maybe that other role be a dependency of this one?